Cargo.toml

```
@@ -57,7 +57,8 @@ staticfile = { version = "0.4", features = [ "cache" ] }
 tempfile = "3.1.0"
 
 # Templating
-tera = { version = "1.3.0", features = ["builtins"] }
+tera = { version = "1.3.1", features = ["builtins"] }
```

```suggestion
tera = { version = "1", features = ["builtins"] }
```
Cargo ignores everything but semver, this makes it look like tera was updated when it wasn't (well, it was updated in the PR but only because you ran `cargo update -p tera`)

---

No, we need at least 1.3.1, as [some generics parameters we need were introduced in that release](https://github.com/Keats/tera/commit/f7835a2c7e2a4b0c25947b70aea3d9327f54eb29). Dunno why it was done in a patch release.

---

Oh weird. Please say that in Cargo.toml then:
```suggestion
tera = { version = ">=1.3.1", features = ["builtins"] }
```

---

I know it's a total style thing, but I think that having a "fully qualified" versions is just better style since it allows programmers to find the correct version of documentation much quicker and is overall more specific than a blanket "give me a `1`". Additionally those two have different meanings, since `1` is asking for tera version `>=1.0.0, <2.0.0` while `1.3.1` is asking for tera version `>=1.3.1, <2.0.0`. Information is lost, and while that in practice has no effect I still think it's not good

---

@jyn514 isn't that equivalent?

---

They are not equivalent. `"1.3.1"` has exactly the same meaning as `"1"`.

---

@jyn514 no, [according to the docs](https://doc.rust-lang.org/stable/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio) `1.3.1` is equivalent to `^1.3.1`, which is equivalent to `>=1.3.1, <2.0.0`. Using just `>=1.3.1` would allow to depend on Tera `2.0.0`, which is not what we want.

---

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

> The string "0.1.12" is a semver version requirement. Since this string does not have any operators in it, it is interpreted the same way as if we had specified "^0.1.12", which is called a caret requirement.

---

No, unadorned versions are [interpreted as careted](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html)

---

I missed that `^0.1.12` has an implicit `>=0.1.12` bound. So this is fine.